org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend

```diff
@@ -110,6 +112,13 @@ import org.eclipse.xtext.validation.Issue
 		if (params.rootPath === null) {
 			throw new IllegalArgumentException("Bad initialization request. rootPath must not be null.")
 		}
+		val rootPath = Paths.get(params.rootPath)
+		if (!Files.exists(rootPath) || !Files.isDirectory(rootPath)) {
+			throw new IllegalArgumentException("Bad initialization request. rootPath does not exist or is not a directory.")
```

Comment:
Sure. Would you take care on that?
Meanwhile I'm too far away from that topic. You can just force push the branch.